Fix/cast details animation and handle missing biography birth info

skaik-mo opened the pull request:
https://github.com/user-attachments/assets/ecc46f36-f228-456b-b6e1-7e3c5d5d0957

skaik-mo committed:
Conditionally render InfoSection and GallerySection

The InfoSection and GallerySection composables will now only be rendered if their respective `description` and `imageUrls` are not empty. This prevents displaying empty sections in the UI.

skaik-mo committed:
Refactor CastDetailsScreen to use Column instead of LazyColumn

The CastDetailsScreen was refactored to use a Column with verticalScroll instead of a LazyColumn. This change was made to simplify the layout and improve performance for smaller lists of items.

Additionally, a Spacer was added at the bottom of the screen to ensure that content is not obscured by the system navigation bar. The height of this spacer is dynamically calculated based on the content height and window height.

The InfoSection composable was also updated, but the implementation details are not yet complete.

K-M-9 commented on designsystem/src/main/java/com/giraffe/designsystem/composable/InfoSection.kt:
```diff
@@ -19,6 +19,7 @@ fun InfoSection(
     descriptionLimits: Int = 100,
     spaceBetween: Int = 8
 ) {
+    if (description.isEmpty()) return
```
You can write it as `if (!description.isEmpty()) { section }`.

skaik-mo replied:
Returning an empty first state is simple and readable, but since this component has no description, displaying only the title is not ideal.

K-M-9 commented on presentation/details/src/main/java/com/giraffe/details/components/gallery/GallerySection.kt:
```diff
@@ -20,6 +20,7 @@ fun GallerySection(
     imageUrls: List<String?>,
     onShowMoreClick: () -> Unit,
 ) {
+    if (imageUrls.isEmpty()) return
```
You can write it as `if (imageUrls.isEmpty()){section}`.

skaik-mo replied:
Returning an empty first state is simple and readable.

K-M-9 commented on presentation/details/src/main/java/com/giraffe/details/screens/castDetails/CastDetailsScreen.kt:
```diff
@@ -53,6 +59,7 @@ fun CastDetailsScreen(
     modifier: Modifier = Modifier,
     castDetailsViewModel: CastDetailsViewModel = koinViewModel(parameters = { parametersOf(personId) })
 ) {
```
Use the `SavedStateHandle` within your `ViewModel` instead of passing values through its constructor.

skaik-mo replied:
We don’t pass arguments directly in the `composable` function. Could you provide an example of how to use `SavedStateHandle` with a `@Serializable `data class?

skaik-mo committed:
integrate saved state handle in cast details view model

- Updated `CastDetailsViewModel` to use `SavedStateHandle` for retrieving `personId`.
- Modified `viewModelModule` to provide `SavedStateHandle` to `CastDetailsViewModel`.
- Adjusted `DetailsApiImp` to remove `personId` argument from `CastDetailsScreen` calls.
- Updated `CastDetailsScreen` to remove `personId` parameter and adjust `koinViewModel` call.

skaik-mo committed:
Merge branch 'develop' into fix/cast-details-animation-and-handle-missing-biography-birth-info

* develop: (30 commits)
  fix series unknown date
  fix series unknown date
  remove modifier from feature api to make each feature independent
  Refactor: Remove unnecessary guest session creation in app initialization
  optimize imports
  Refactor: Enable resource shrinking and minification for release build
  handle cross-feature navigation state by creating back stack
  remove unneeded comments
  remove debug code logs
  handle back buttons click
  movie injecting apis outside on create
  handle sending invalid cast id instead of default value
  rename app navigator and app screen
  fix: copy text with new line and convert it in single line
  Refactor: Migrate Explore module to use type-safe navigation
  Refactor DetailsApi and NavGraph
  clean up unused imports
  fix: remove semantic modifier
  fix: add single line to search field and make max characters 100
  clean up unused imports
  ...

# Conflicts:
#	presentation/details/src/main/java/com/giraffe/details/navigation/DetailsApiImp.kt
#	presentation/details/src/main/java/com/giraffe/details/screens/castDetails/CastDetailsScreen.kt
#	presentation/details/src/main/java/com/giraffe/details/screens/castDetails/CastDetailsViewModel.kt
